docs/development/submitting-patches.rst

```diff
@@ -57,6 +57,72 @@ Important information should be documented with prose in the ``docs`` section.
 To ensure it builds and passes `doc8`_ style checks you can run
 ``make docs`` and ``make lint`` respectively.
 
+Keeping your local branch updated
+---------------------------------
+
+As you work, you will need to keep your local ``master`` branch up-to-date with
+the ``master`` branch in the main Warehouse repository, which moves forward as
+the maintainers merge pull requests. Most people working on the project use
+this workflow.
+
+The following documentation assumes that you have git configured so that when
+you run the following command:
+
+.. code-block:: console
+
+  git remote -v
+
+Your output looks like this:
+
+.. code-block:: console
+
+  origin  https://github.com/USERNAME/warehouse.git (fetch)
+  origin  https://github.com/USERNAME/warehouse.git (push)
+  upstream  https://github.com/pypa/warehouse.git (fetch)
+  upstream  https://github.com/pypa/warehouse.git (push)
+
+
+In the example above, ``USERNAME`` is your username on GitHub.
+
+First, fetch the latest changes from the main Warehouse repository,
+``upstream``:
+
+.. code-block:: console
+
+  git fetch upstream
+
+Then, check out your local ``master`` branch, and rebase the changes on top of
+it:
+
+.. code-block:: console
+
+  git checkout master
+  git rebase upstream/master
+
+Next, push the updates you have just made to your local ``master`` branch to
+your ``origin`` repository on GitHub:
+
+.. code-block:: console
+
+  git checkout master
+  git push origin master
+
+Now your local ``master`` branch and the ``master`` branch in your ``origin``
+repo have been updated with the most recent changes from the main Warehouse
+repository.
+
+To keep your feature branches updated, the process is similar:
+
+.. code-block:: console
+
+   git checkout awesome-feature
+   git fetch upstream
+   git rebase upstream/master
```

Comment:
Might be worth mentioning the next step here as well: force-pushing the branch to the remote.

Reply:
Do you mean force-pushing the feature branch to ``origin`` while you are working?  Or for making a pull request, or both?

Reply:
Just to keep your branch up-to-date on your remote, whether you've already made a PR or not.

The `master` branch here moves somewhat quickly due to all the dependency upgrades that get merged, so it's pretty likely that a new contributor will have to force-push at least once while working on their pull request.